include/Gaffer/MetadataAlgo.h

```diff
@@ -153,6 +154,10 @@ GAFFER_API void copy( const GraphComponent *from, GraphComponent *to, const IECo
 /// \undoable
 GAFFER_API void copyColors( const Gaffer::Plug *srcPlug, Gaffer::Plug *dstPlug, bool overwrite );
 
+/// Some metadata can cause clashes if respective entries exist in the scene already.
+GAFFER_API bool metadataRegistrationCanClash( const IECore::InternedString &key );
```

Comment:
This doesn't feel quite right to me, but I'm struggling to rationalise/articulate why. I think a part of it is just the ambiguity of "can clash". What is a clash? Is it saying that it is allowed to clash, or that it isn't, or that it might and you need to deal with it?

But I think perhaps the bigger thing is that we already have special methods for registering numeric bookmarks, and now we're introducing another set of special methods that you need to use instead in certain circumstances. If we want to make the "non clashing metadata" concept watertight, then perhaps it would need to be a feature of `Metadata` rather than `MetadataAlgo`?